src/macro/eval/evalMain.ml

```diff
@@ -436,8 +439,52 @@ let rec value_to_expr v p =
 				let args = List.map (fun v -> value_to_expr v p) (Array.to_list e.eargs) in
 				(ECall (epath, args), p)
 		end
-
-	| _ -> exc_string ("Cannot convert " ^ (value_string v) ^ " to expr")
+	| VInstance {ikind = IStringMap smap} ->
+		begin
+			let decl = ref [] in
+			let last_value = ref None in
+			try
```

Comment:
The pattern here is strange. Why have the `try` outside the `iter` if you want to have error messages that are specific to elements? You can just put it inside so you don't have to deal with this `last_value` stuff.

Reply:
okay ocalm it a bit different as other languages.